executor/write.go

```diff
@@ -190,28 +172,30 @@ func updateRecord(ctx context.Context, sctx sessionctx.Context, h kv.Handle, old
 
 	// 5. If handle changed, remove the old then add the new record, otherwise update the record.
 	if handleChanged {
-		if sc.DupKeyAsWarning {
-			// For `UPDATE IGNORE`/`INSERT IGNORE ON DUPLICATE KEY UPDATE`
-			// If the new handle or unique index exists, this will avoid to remove the record.
-			err = tables.CheckHandleOrUniqueKeyExistForUpdateIgnoreOrInsertOnDupIgnore(ctx, sctx, t, newHandle, newData, modified)
+		if updated, err := func() (bool, error) {
+			txn, err := sctx.Txn(true)
 			if err != nil {
-				if terr, ok := errors.Cause(err).(*terror.Error); sctx.GetSessionVars().StmtCtx.IgnoreNoPartition && ok && terr.Code() == errno.ErrNoPartitionForGivenValue {
-					return false, nil
-				}
 				return false, err
 			}
-		}
-		if err = t.RemoveRecord(sctx, h, oldData); err != nil {
-			return false, err
-		}
-		// the `affectedRows` is increased when adding new record.
-		if sc.DupKeyAsWarning {
-			_, err = t.AddRecord(sctx, newData, table.IsUpdate, table.SkipHandleCheck, table.WithCtx(ctx))
-		} else {
+			memBuffer := txn.GetMemBuffer()
```

Comment:
We could add some comments like L194 does for the `IGNORE` path.

Reply:
What do you think the comment should be?

Reply:
For example
```
// For `UPDATE IGNORE`/`INSERT IGNORE ON DUPLICATE KEY UPDATE` statement,
// using the stating buffer so that we don't need to precheck the existence of handle or unique keys with 
// extra kv requests, and the remove will not take effect if there are conflicts.
```

Reply:
done